### ISSUE TITLE
"Unexpected error" when showing dashboard with Public role for non-logged in user, while it works for logged in users with Public role

### ISSUE THREAD

**Issue by akelai** [apache/superset#16382]
If I try to access a dashboard as a non-logged in user, I get an "Unexpected error".
But if I try to access the same dashboard as a logged-in user, with a user who has "Public" as the only role assigned, it works.

### Expected results

I have the line `PUBLIC_ROLE_LIKE = "Gamma"` in my `superset_config.py`, so the "Public" role should allow a non-logged in user to view the dashboard.

### Actual results

As a non-logged in user, when I go to the dashboard url, I get an "Unexpected error" immediately after the top bar, where the dashboard should be loaded. Expanding the error message with "See more", I get:
```
Unexpected error: TypeError: Cannot convert undefined or null to object

    in lh
    in d
    in uh
    in T
    in m
    in Unknown
    in s
    in fh
```
If I click the "Copy message" button, I get this in the clipboard: `[object Object]`.

#### How to reproduce the bug

1. Have `PUBLIC_ROLE_LIKE = "Gamma"` in `superset_config.py` (and run `superset init` if it wasn't there since the beginning)
2. Login as admin and create a dashboard
3. Copy the dashboard's url
4. Logout
5. Paste the dashboard url in browser and load the page
6. The error appears

Instead, do the following to see what happens (no problem) when using a logged-in user with only "Public" as an associated role:

1. Have `PUBLIC_ROLE_LIKE = "Gamma"` in `superset_config.py` (and run `superset init` if it wasn't there since the beginning)
2. Login as admin and create a dashboard
3. Create a test user and assign the "Public" role to it
4. Copy the dashboard's url
5. Logout
6. Login as the test user with the "Public" role
7. Paste the dashboard url in browser and load the page
8. The dashboard appears

### Environment

(please complete the following information):

- superset version: 1.2.0
- python version: 3.7.10
- node.js version: 12.16.1

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

### Additional context

I've found [this issue](https://github.com/apache/superset/issues/14442) reporting some similar problem, but the error is a bit different, the Superset version was older, and the issue had been closed thanks to a fix in the Superset code, while I have installed a fresh Superset 1.2.0 and experiencing this explained problem.

**Comment by akelai** [comment]
Problem solved after an upgrade to version 1.3.0:
```shell
pip install apache-superset==1.3.0 --upgrade
superset db upgrade
superset init
```

**Comment by panqingdian** [comment]
Will this be fixed with the 1.3.0 update
